build/scraper.mjs

```diff
@@ -13,47 +11,65 @@ import WarframeScraper from './wikia/scrapers/WarframeScraper.mjs';
 import VersionScraper from './wikia/scrapers/VersionScraper.mjs';
 import readJson from './readJson.mjs';
 
+import { get, getJSON, retryAttempts } from './network.mjs';
+
 const locales = await readJson(new URL('../config/locales.json', import.meta.url));
 
 const prod = process.env.NODE_ENV === 'production';
-// eslint-disable-next-line no-control-regex
-const sanitize = (str) => str.replace(/\\r|\r?\n|\x09/g, '').replace(/\\\\"/g, "'");
-const agent = process.env.SOCK5_HOST
-  ? new Agent({
-      socksHost: process.env.SOCKS5_HOST,
-      socksPort: process.env.SOCKS5_PORT,
-      socksUsername: process.env.SOCKS5_USER,
-      socksPassword: process.env.SOCKS5_PASS,
-    })
-  : undefined;
-const get = async (url, disableProxy = !prod, compress = false) => {
-  const res = await fetch(url, {
-    agent: disableProxy ? undefined : agent,
-  });
-  return compress === false ? Uint8Array.from(await res.buffer()) : res.text();
-};
-const getJSON = async (url, disableProxy) => JSON.parse(sanitize(await get(url, disableProxy, true)));
 
 /**
  * Retrieves the base item data necessary for the parsing process
  */
 class Scraper {
+  endpointCache = new Map();
+
+  originServerAvailable = false;
+
+  async checkOriginServerAvailability() {
+    this.originServerAvailable = true;
+    try {
+      // If this call is successful, it means that the origin server is available.
+      await this.fetchEndpoints(true, 'en');
+    } catch (err) {
+      // Origin server not available, fall back to content server
+      this.originServerAvailable = false;
```

Comment:
seems redundant since default is false.

Reply:
It is not, originServerAvailable is set to true at the beginning so that fetchEndpoints (line 32) uses the origin server for the check. So if it fails, you need to set it to false

Reply:
ah, gotcha now. much appreciated.